src/content/docs/en/guides/assets.mdx

```diff
@@ -0,0 +1,213 @@
+
+```ts
+interface ImageAsset {
+  src: string;
+  width: number;
+  height: number;
+  format: string;
+}
+```
+
+## New `src/assets` directory
+
+We recommend putting images in the `src/assets` directory. This is entirely optional, but will become the recommended location for other asset types in the future.
+
+Your images stored in `src/assets/` can be used by components (`.astro`, `.mdx` and other UI frameworks) and in Markdown files.
+
+Import them from a **relative file path** or [import alias](/en/guides/aliases/) in any component file and then use the import as the `Image` component's `src` attribute.
+
+```astro
+---
+// src/pages/index.astro
+import { Image } from 'astro:assets';
+
+// Access images in `src/images/`
+import logo from '../assets/images/logo.png';
+---
+<Image src={logo} alt="Astro" />
+```
+
+## New Markdown support
+
+Relative images can now be referenced in Markdown and MDX files. These will automatically be optimized and hashed by Astro's build process.
+
+Remote images are also valid, but are not automatically optimized by Astro's build process.
+
+```md
+<!-- src/pages/post-1.md -->
+# My Markdown Page
+
+<!-- Local image stored at src/assets/stars.png -->
+![A starry night sky.](../../assets/stars.png)
+<img src="/_assets/stars.hash.png" alt="A starry night sky.">
+
+<!-- Remote image on another server -->
+![Astro](https://astro.build/assets/logo.png)
+<img src="https://astro.build/assets/logo.png" width="25" alt="Astro">
+```
+
+### Content Collections
+
+A common use case for blog posts is to reference assets in their metadata (think, cover image for an article, picture for an author etc).
+
+With the `src/assets` folder, we're also introducing a new `asset` helper to the `astro:content` module. The `asset` helper will validate that the user references a valid asset from the `src/assets` folder:
+
+```ts
+import { asset, defineCollection, z } from "astro:content";
+
+const blogCollection = defineCollection({
+  schema: z.object({
+    title: z.string(),
+    image: asset({ width: 1120, height: 1120 }),
+  }),
+});
+```
+
+Assets referred this way will be transformed automatically to the same shape as if the image was imported, so they're ready to be passed to the `Image` component or the `getImage` function.
+
+## Features
+
+Enabling asset support allows you to access the `astro:assets` module. This module exposes the following features:
+
+- `Image` (see [Image Component](#image-component))
+
+### Image Component
+
+The `<Image>` component generates an `<img>` tag with all the necessary attributes needed to display the image while avoiding Content Layout Shift **(CLS)**.
+
+:::tip[What is Content Layout Shift?]
+CLS is a score that reflects how much content shifted on your page during loading. The ideal number for CLS is as close to zero as possible.
```

Comment:
What does the Image component do to prevent CLS? We should add that here

Reply:
```suggestion
CLS is a score that reflects how much content shifted on your page during loading. The ideal number for CLS is as close to zero as possible, which the `Image` component enforces by automatically selecting the correct `width` and `height` for local images.
```